lib/faraday/error.rb

```diff
@@ -6,10 +6,11 @@ module Faraday
   class Error < StandardError
     attr_reader :response, :wrapped_exception
 
-    def initialize(exc, response = nil)
+    def initialize(exc, deprecated_response = nil, response = nil)
```

Comment:
Let me assume that in faraday v2 this signature should be changed to `def initialize(msg = nil, response:)` (as I mentioned in the PR description). Then, first I thought to change the signature at this time to such like

```rb
def initialize(exc, deperecated_response = nil, response: nil)
```

but I couldn't to do that because the old response object is hash and it conflicts with the keyword argument.